src/main/java/io/confluent/examples/streams/ApplicationResetExample.java

```diff
@@ -143,32 +144,47 @@ public static void main(final String[] args) {
     streamsConfiguration.put(ConsumerConfig.AUTO_OFFSET_RESET_CONFIG, "earliest");
 
     final boolean doReset = args.length > 1 && args[1].equals("--reset");
-    final KafkaStreams streams = run(doReset, streamsConfiguration);
+    final KafkaStreams streams = buildKafkaStreams(streamsConfiguration);
 
     // Add shutdown hook to respond to SIGTERM and gracefully close Kafka Streams
     Runtime.getRuntime().addShutdownHook(new Thread(streams::close));
+
+    // Delete the application's local state on reset
+    if (doReset) {
+      streams.cleanUp();
+    }
+
+    startKafkaStreams(streams);
   }
 
-  public static KafkaStreams run(final boolean doReset, final Properties streamsConfiguration) {
+  public static KafkaStreams buildKafkaStreams(final Properties streamsConfiguration) {
     // Define the processing topology
     final StreamsBuilder builder = new StreamsBuilder();
     final KStream<String, String> input = builder.stream("my-input-topic");
     input.selectKey((key, value) -> value.split(" ")[0])
-      .groupByKey()
-      .count()
-      .toStream()
-      .to("my-output-topic", Produced.with(Serdes.String(), Serdes.Long()));
+         .groupByKey()
+         .count()
+         .toStream()
+         .to("my-output-topic", Produced.with(Serdes.String(), Serdes.Long()));
 
-    final KafkaStreams streams = new KafkaStreams(builder.build(), streamsConfiguration);
+    return new KafkaStreams(builder.build(), streamsConfiguration);
+  }
 
-    // Delete the application's local state on reset
-    if (doReset) {
-      streams.cleanUp();
-    }
+  public static void startKafkaStreams(final KafkaStreams streams) {
```

Comment:
nit: `startKafkaStreamsSync` (or similar -- otherwise unclear what the difference to calling `#start()`

Reply:
Ok.